src/analysis/seminorm.lean

```diff
@@ -584,6 +590,59 @@ lemma comp_smul_apply (p : seminorm 𝕜 F) (f : E →ₗ[𝕜] F) (c : 𝕜) (x
 
 end semi_normed_comm_ring
 
+section normed_field
+variables [normed_field 𝕜] [add_comm_group E] [module 𝕜 E]
+
+private lemma bdd_below_range_add (x : E) (p q : seminorm 𝕜 E) :
+  bdd_below (range (λ (u : E), p u + q (x - u))) :=
+by { use 0, rintro _ ⟨x, rfl⟩, exact add_nonneg (p.nonneg _) (q.nonneg _) }
+
+noncomputable instance : has_inf (seminorm 𝕜 E) :=
+{ inf := λ p q,
+  { to_fun := λ x, ⨅ u : E, p u + q (x-u),
+    triangle' := λ x y, begin
+      refine le_cinfi_add_cinfi (λ u v, _),
+      apply cinfi_le_of_le (bdd_below_range_add _ _ _) (v+u), dsimp only,
+      convert add_le_add (p.triangle v u) (q.triangle (y-v) (x-u)) using 1,
+      { rw show x + y - (v + u) = y - v + (x - u), by abel },
+      { abel },
+    end,
+    smul' := λ a x, begin
```

Comment:
Are there some lemmas about `a * ⨅ i, f i` and `⨅ i, a * f i` hiding in this proof?

Reply:
This is a bit shorter:
```lean
    smul' := λ a x, begin
      obtain ha | ha := (norm_nonneg a).eq_or_lt,
      { rw ← ha, ring_nf, apply le_antisymm,
        { apply cinfi_le_of_le (bdd_below_range_add _ _ _) (0:E),
          simp only [seminorm.zero, sub_zero, zero_add, q.smul, ← ha, zero_mul] },
        { exact le_cinfi (λ u, add_nonneg (p.nonneg _) (q.nonneg _)) } },
      change _ = order_iso.mul_left₀ (∥a∥) ha _,
      rw order_iso.map_cinfi _ (bdd_below_range_add x _ _),
      dsimp,
      simp_rw mul_add,
      have : a ≠ 0 := sorry,
      have h : function.right_inverse  (λ x : E, a • x) (λ x : E, a⁻¹ • x) := inv_smul_smul₀ this,
      apply infi_congr (λ x : E, a⁻¹ • x) h.surjective,
      simp [←seminorm.smul, smul_sub, smul_inv_smul₀ this],
    end
```